test/integration/scheduler_perf/util.go

```diff
@@ -165,56 +165,56 @@ func (*metricsCollector) run(ctx context.Context) {
 
 func (pc *metricsCollector) collect() []DataItem {
 	var dataItems []DataItem
-	for _, metric := range pc.Metrics {
-		dataItem := collectHistogram(metric, pc.labels)
-		if dataItem != nil {
-			dataItems = append(dataItems, *dataItem)
-		}
+	for metric, wantLabel := range pc.MetricLabelMap {
+		itemCollection := collectHistogram(metric, pc.labels, wantLabel)
+		dataItems = append(dataItems, itemCollection...)
 	}
 	return dataItems
 }
 
-func collectHistogram(metric string, labels map[string]string) *DataItem {
-	hist, err := testutil.GetHistogramFromGatherer(legacyregistry.DefaultGatherer, metric)
+func collectHistogram(metric string, labels map[string]string, wantLabel string) []DataItem {
+	var dataItems []DataItem
+	histMap, err := testutil.GetHistogramFromGatherer(legacyregistry.DefaultGatherer, metric, wantLabel)
 	if err != nil {
 		klog.Error(err)
 		return nil
 	}
-	if hist.Histogram == nil {
-		klog.Errorf("metric %q is not a Histogram metric", metric)
-		return nil
-	}
-	if err := hist.Validate(); err != nil {
-		klog.Error(err)
-		return nil
-	}
+	for label, hist := range histMap {
+		if err := hist.Validate(); err != nil {
+			klog.Error(err)
+			return nil
+		}
 
-	q50 := hist.Quantile(0.50)
-	q90 := hist.Quantile(0.90)
-	q99 := hist.Quantile(0.95)
-	avg := hist.Average()
+		q50 := hist.Quantile(0.50)
+		q90 := hist.Quantile(0.90)
+		q99 := hist.Quantile(0.95)
+		avg := hist.Average()
 
-	// clear the metrics so that next test always starts with empty prometheus
-	// metrics (since the metrics are shared among all tests run inside the same binary)
-	hist.Clear()
+		// clear the metrics so that next test always starts with empty prometheus
+		// metrics (since the metrics are shared among all tests run inside the same binary)
+		hist.Clear()
 
-	msFactor := float64(time.Second) / float64(time.Millisecond)
+		msFactor := float64(time.Second) / float64(time.Millisecond)
 
-	// Copy labels and add "Metric" label for this metric.
-	labelMap := map[string]string{"Metric": metric}
-	for k, v := range labels {
-		labelMap[k] = v
-	}
-	return &DataItem{
-		Labels: labelMap,
-		Data: map[string]float64{
-			"Perc50":  q50 * msFactor,
-			"Perc90":  q90 * msFactor,
-			"Perc99":  q99 * msFactor,
-			"Average": avg * msFactor,
-		},
-		Unit: "ms",
+		// Copy labels and add "Metric" label for this metric.
+		labelMap := map[string]string{"Metric": label}
```

Comment:
I believe `labels` already has the metric name. Better add `{wantLabel: label}`, where label is the value for that label,
like: `{"extension_point": "Filter"}`

Reply:
Unfortunately not, `labels` is defined here, 
https://github.com/kubernetes/kubernetes/blob/656d00e8949f64cd663ca532d12e00982a6664c4/test/integration/scheduler_perf/scheduler_perf_test.go#L593
and pass down to here (the second parameter).

it is something like this
```
BenchmarkPerfScheduling/Preemption/500Nodes/namespace-2
```

`metric` name  is already embedded in `label`, it will result to be, 
```
"Metric": "scheduler_framework_extension_point_duration_seconds/Filter"
```

Reply:
I remember seeing the metric names somewhere in the report when I run this in the past. Could you re-check?

Reply:
ah, sorry, it was the `metric` variable.
Instead of Replacing it with `fmt.Sprintf("%s/%s", metricName, label.GetValue())`, better have them as separate labels in the `labelMap`.

Reply:
That's actually how prometheus preserves metric name under the hood, the metric name is a label value where the label is `__name__`.

Reply:
> Instead of Replacing it with fmt.Sprintf("%s/%s", metricName, label.GetValue()), better have them as separate labels in the labelMap

Done, pls see the PR description for how it looks like now.